### ISSUE TITLE
DatePicker - improve user user experience for keyboard users

### ISSUE THREAD

**Issue by Saegrov** [fremtind/jokul#773]
**Is your feature request related to a problem? Please describe.**
With the current implementation of DatePicker, it opens when the user focuses on the input field. This is problematic for users using tab navigation in two different ways:

1. **Forms with optional date fields.**
When navigating in a form using tab-key in a form that has an optional date field you have to tab through the entire datepicker even though you are not interested in selecting a date. Example: 
![Screen-Recording-2020-02-11-at-10 07 00](https://user-images.githubusercontent.com/124469/74223996-3085f400-4cb8-11ea-9974-a60a99495790.gif)
2. **Typing a date with keyboard**
Using the DatePicker to type the date gives the user two options when leaving the DatePicker. The user either has to use the mouse to click outside somewhere or the user has to use tab navigation to tab through the entire datepicker. Neither option is good for a keyboard user.
![Screen-Recording-2020-02-11-at-10 22 23](https://user-images.githubusercontent.com/124469/74224684-93c45600-4cb9-11ea-8c21-ea2e48e44560.gif)

**Describe the solution you'd like**
The DatePicker should have a separate button to open the datepicker. This would create a good user experience for both keyboard users and mouse users.

**Additional context**
When Jøkul implemented open on focus we reported a bug in NRKs `core-datepicker` where they discussed if this was a good option or if there were any best practices regarding this. They didn't find any best practices but they found that using a button to open was good for them.

See: https://github.com/nrkno/core-components/issues/322
